compiler/rustc_next_trait_solver/src/canonicalizer.rs

```diff
@@ -232,197 +214,213 @@ impl<'tcx> TypeFolder<TyCtxt<'tcx>> for Canonicalizer<'_, 'tcx> {
             // FIXME: We should investigate the perf implications of not uniquifying
             // `ReErased`. We may be able to short-circuit registering region
             // obligations if we encounter a `ReErased` on one side, for example.
-            ty::ReStatic | ty::ReErased => match self.canonicalize_mode {
-                CanonicalizeMode::Input => CanonicalVarKind::Region(ty::UniverseIndex::ROOT),
+            RegionKind::ReStatic | RegionKind::ReErased => match self.canonicalize_mode {
```

Comment:
that is :/ can you instead add `use rustc_type_ir as ty` and then keep using `ty::X` everywhere?

Reply:
The glob-import of all these types don't happen in `rustc_type_ir`, but in `rustc_middle`. I guess I could uplift that.